backend-project/small_eod/migration_v1/README.md

```diff
@@ -0,0 +1,44 @@
+
+###Bootstrap on dev:
+
+To run migrator locally:
+```shell script
+docker-compose up -d migration_db
+docker-compose exec migration_db mysql -ppass --execute 'CREATE DATABASE small_eod CHARACTER SET utf8;'
+docker-compose exec -T migration_db mysql small_eod -ppass < ./small_eod_demo-26-09-2020.sql
+docker-compose up -d --build
+docker-compose run --rm backend python manage.py migrate_v1
+```
+
+### Added unique constraints:
+
+I added some simple constraints that seemed missing from the data model:
+ - Channel.name
+ - Feature.name
+ - FeatureOption.('name', 'feature')
+ - DocumentType.name
+ - Tag.name
+
+### Migration assumptions:
+
+I had to make some assumptions about "what(v1)-goes-where(v2)" that may or may not be accurate:
+ - Dictionary.type is now Feature.name
+ - Dictionary.name is now FeatureOption.name
+ - LetterName model is now DocumentType
+ - Users responsible for case will want be notified
+ - Letter.identifier is now Letter.reference_number
+ 
+### Issues:
+ - v1 has no data for created_by and modified_by fields in general
+ - Case.comment field too short for source data
+ - Should Feature.name be translated?
+ - Should we subtract 1 while migrating Letter ordering?
+ - When migrating tags, should something be done with TagNamespace as well?
+ - There is no person model in v2 as it relies on users.User directly and
+   relation with auth.User in v1 is not enforced, so we cannot be sure if we have all source Users.
+   There are no Person-model auth.Users in the demo dump.
```

Comment:
users_users should be used instead of `auth.Users` thanks to swappable auth user model. See details https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#substituting-a-custom-user-model .

Reply:
In v2 yes, but there are no `users.User` in v1 - or did I miss it?

Reply:
Person - available in v1, dropped in v2
User - build-in in v1 (see https://github.com/watchdogpolska/small_eod/blob/master/small_eod/users/models.py ), swappable in v2 (https://github.com/watchdogpolska/small_eod/blob/dev/backend-project/small_eod/users/models.py)

`auth.Users` from v1 match to `users.User` from v2. See class inheritance of `users.User`.

We need recreate relationship from Person to User in v1, but we can handle that.